phoenix-tools/src/main/java/org/apache/phoenix/schema/SchemaSynthesisProcessor.java

```diff
@@ -154,7 +156,24 @@ private CreateTableStatement getCreateTableStatement(AddColumnStatement alterSta
         } else {
             newColDef.addAll(oldColDef);
             newColDef.addAll(addStmt.getColumnDefs());
-            newCreateStmt = new CreateTableStatement(createStmt, newColDef);
+            PrimaryKeyConstraint oldPKConstraint = createStmt.getPrimaryKeyConstraint();
+            List<ColumnDefInPkConstraint> pkList = new ArrayList<>();
+            for(Pair<ColumnName, SortOrder> entry : oldPKConstraint.getColumnNames()) {
+                ColumnDefInPkConstraint cd = new
+                        ColumnDefInPkConstraint(entry.getFirst(), entry.getSecond(), oldPKConstraint.isColumnRowTimestamp(entry
+                .getFirst()));
+                pkList.add(cd);
+            }
+            for(ColumnDef cd : addStmt.getColumnDefs()) {
```

Comment:
What if the ALTER tries to add an existing PK column?

Reply:
That's a valid point, I am thinking to create a new Jira which will validate if the statements are valid on the mini-cluster and then only proceed for synthesis. Does that sound okay?